src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Data.Common;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Models;
+using Microsoft.Data.SqlClient;
+
+namespace Azure.DataApiBuilder.Core.Resolvers
+{
+    /// <summary>
+    /// Class for building DwSql queries.
+    /// </summary>
+    public class DWSqlQueryBuilder : BaseSqlQueryBuilder, IQueryBuilder
+    {
+
+        private static DbCommandBuilder _builder = new SqlCommandBuilder();
+
+        /// <inheritdoc />
+        public override string QuoteIdentifier(string ident)
+        {
+            return _builder.QuoteIdentifier(ident);
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlQueryStructure structure)
+        {
+            string columns;
+            string stringAgg = string.Empty;
+            int i = 0;
+            // Iterate through all the columns and build the string_agg
+            foreach (LabelledColumn column in structure.Columns)
+            {
+                // Generate the col value.
+                bool subQueryColumn = structure.IsSubqueryColumn(column);
+                string col_value = column.Label;
+
+                // If the column is not a subquery column and is not a string, cast it to string
```

Comment:
why do we want to cast this to a string when it isn't one already and not a subquery column?

Reply:
This is because STRING_AGG which we are using to create the json requires strings. Hence all other data types need to be cast into a string.